.github/workflows/build-on-pr.yml

```diff
@@ -5,7 +5,7 @@ on:
 
 permissions:
   contents: read
-  security-events: read  # This is required to handle authentication to our artifact publishing API
+  id-token: write # This is required to handle authentication to our artifact publishing API
```

Comment:
This is the only tricky one to review, but here the triggering event is a pull_request, and the workflow definition being used is the one in the main branch.